dataprofiler/tests/profilers/test_profile_builder.py

```diff
@@ -749,6 +752,394 @@ def test_min_max_id_properly_update(self, *mocks):
+        self.assertEqual(4, profiler.total_samples)
+        self.assertEqual(1, profiler._empty_line_count)
+        self.assertEqual("<class 'pandas.core.series.Series'>",
+                         profiler.file_type)
+        self.assertIsNone(profiler.encoding)
+        self.assertIsInstance(profiler._profile, UnstructuredCompiler)
+
+        # can properties update correctly for data loaded from file
+        data = pd.Series(['this', 'is my', '\n\r', 'test'])
+        mock_data_reader = mock.Mock(spec=dp.data_readers.csv_data.CSVData)
+        mock_data_reader.data = data
+        mock_data_reader.data_type = 'csv'
+        mock_data_reader.file_encoding = 'utf-8'
+        mock_data_reader.input_file_path = 'fake/path/file.csv'
+
+        profiler = UnstructuredProfiler(mock_data_reader)
+        self.assertEqual(4, profiler.total_samples)
+        self.assertEqual(1, profiler._empty_line_count)
+        self.assertEqual("csv", profiler.file_type)
+        self.assertEqual("utf-8", profiler.encoding)
+        self.assertIsInstance(profiler._profile, UnstructuredCompiler)
+
+    def test_merge_profiles(self, *mocks):
+        # can properties update correctly for data
+        data1 = pd.Series(['this', 'is my', '\n\r', 'test'])
+        data2 = pd.Series(['here\n', '\t    ', ' ', ' is', '\n\r', 'more data'])
+
+        # create profilers
+        profiler1 = UnstructuredProfiler(data1)
+        profiler2 = UnstructuredProfiler(data2)
+
+        # mock out _profile
+        profiler1._profile = 1
+        profiler2._profile = 2
+
+        # merge profilers
+        merged_profile = profiler1 + profiler2
+        self.assertEqual(10, merged_profile.total_samples)
+        self.assertEqual(4, merged_profile._empty_line_count)
+        # note how sample doesn't include whitespace lines
+        self.assertCountEqual(['this', ' is', 'here\n', 'more data', 'is my'],
+                              merged_profile.sample)
+        self.assertEqual(3, merged_profile._profile)
+
+    def test_get_sample_size(self, *mocks):
+        data = pd.DataFrame([0] * int(50e3))
+
+        # test data size < min_sample_size = 5000 by default
+        profiler = UnstructuredProfiler(None)
+        profiler._min_sample_size = 5000
+        profiler._sampling_ratio = 0.2
+        sample_size = profiler._get_sample_size(data[:1000])
+        self.assertEqual(1000, sample_size)
+
+        # test data size * 0.20 < min_sample_size < data size
+        sample_size = profiler._get_sample_size(data[:10000])
+        self.assertEqual(5000, sample_size)
+
+        # test min_sample_size > data size * 0.20
+        sample_size = profiler._get_sample_size(data)
+        self.assertEqual(10000, sample_size)
+
+        # test min_sample_size > data size * 0.10
+        profiler._sampling_ratio = 0.5
+        sample_size = profiler._get_sample_size(data)
+        self.assertEqual(25000, sample_size)
+
+    def test_clean_data_and_get_base_stats(self, *mocks):
```

Comment:
Can you leave comments about what got sampled in the cases

Reply:
updated